id-repository/id-repository-core/src/main/java/io/mosip/idrepository/core/manager/CredentialServiceManager.java

```diff
@@ -392,7 +406,7 @@ public void sendRequestToCredService(List<CredentialIssueRequestDto> eventReques
 			BiConsumer<CredentialIssueRequestWrapperDto, Map<String, Object>> credentialRequestResponseConsumer) {
 		eventRequestsList.forEach(reqDto -> {
 			CredentialIssueRequestWrapperDto requestWrapper = new CredentialIssueRequestWrapperDto();
-			requestWrapper.setRequest(reqDto);
+			requestWrapper.setRequest((CredentialRequestV2DTO) reqDto);
```

Comment:
why we need casting, cannot change the declaration of the eventRequestsList itself to take V2 DTO or can we not directly do the changes in CredentialIssueRequestDto ? Asking this because at the service end it is already only the old object
 https://github.com/mosip/id-repository/blob/develop/id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/controller/CredentialRequestGeneratorController.java#L116
Casting everywhere is makes the following of the code difficult